be/src/exec/parquet_scanner.cpp

```diff
@@ -133,13 +127,21 @@ Status ParquetScanner::open_file_reader() {
             break;
         }
 #endif
-        default: {
-            std::stringstream ss;
-            ss << "Unknown file type, type=" << range.file_type;
-            return Status::InternalError(ss.str());
+            default: {
+                std::stringstream ss;
+                ss << "Unknown file type, type=" << range.file_type;
+                return Status::InternalError(ss.str());
+            }
+        }
+        RETURN_IF_ERROR(file_reader->open());
```

Comment:
If reader open fail, file_reader will be a memory leak. you should use std::unique_ptr<>